code generation problem:  variable [id] is already defined in class

There is a problem with code generation from schema, in this case when there is a table field name that has the same name that the PK, but i suppose that this can happen with others too.

Table DDL:

``` sql
CREATE TABLE [dbo].[HOTEL_TIPO_HAB_OVERBOOKING](
    [ID] [int] IDENTITY(1,1) NOT NULL,
    [FK_ID_HOTEL_MOVERBOOKING] [int] NOT NULL,
    [FK_CODIGOHOT] [varchar](3) NOT NULL,
    [FK_CODTIPOHAB] [varchar](6) NOT NULL,
    [CANTIDAD] [int] NOT NULL,
    [FECHA] [datetime] NOT NULL,
 CONSTRAINT [ID] PRIMARY KEY NONCLUSTERED 
(
    [ID] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]

GO

SET ANSI_PADDING OFF
GO

ALTER TABLE [dbo].[HOTEL_TIPO_HAB_OVERBOOKING]  WITH NOCHECK ADD  CONSTRAINT [FK_HOTEL_TIPO_HAB_OVERBOOKING_HOTEL_MOVERBOOKING] FOREIGN KEY([FK_ID_HOTEL_MOVERBOOKING])
REFERENCES [dbo].[HOTEL_MOVERBOOKING] ([ID])
GO

ALTER TABLE [dbo].[HOTEL_TIPO_HAB_OVERBOOKING] CHECK CONSTRAINT [FK_HOTEL_TIPO_HAB_OVERBOOKING_HOTEL_MOVERBOOKING]
GO

ALTER TABLE [dbo].[HOTEL_TIPO_HAB_OVERBOOKING]  WITH NOCHECK ADD  CONSTRAINT [FK_HOTEL_TIPO_HAB_OVERBOOKING_HOTEL_TIPO_HAB] FOREIGN KEY([FK_CODTIPOHAB], [FK_CODIGOHOT])
REFERENCES [dbo].[HOTEL_TIPO_HAB] ([CODTIPOHAB], [FK_CODIGOHOT])
GO

ALTER TABLE [dbo].[HOTEL_TIPO_HAB_OVERBOOKING] CHECK CONSTRAINT [FK_HOTEL_TIPO_HAB_OVERBOOKING_HOTEL_TIPO_HAB]
GO
```

Generated Querydsl query type for HotelTipoHabOverbooking

``` java
@Generated("com.mysema.query.sql.codegen.MetaDataSerializer")
public class QHotelTipoHabOverbooking extends com.mysema.query.sql.RelationalPathBase<HotelTipoHabOverbooking> {

    private static final long serialVersionUID = -1385408739;

    public static final QHotelTipoHabOverbooking hotelTipoHabOverbooking = new QHotelTipoHabOverbooking("HOTEL_TIPO_HAB_OVERBOOKING");

    public final NumberPath<Integer> cantidad = createNumber("cantidad", Integer.class);

    public final DateTimePath<java.sql.Timestamp> fecha = createDateTime("fecha", java.sql.Timestamp.class);

    public final StringPath fkCodigohot = createString("fkCodigohot");

    public final StringPath fkCodtipohab = createString("fkCodtipohab");

    public final NumberPath<Integer> fkIdHotelMoverbooking = createNumber("fkIdHotelMoverbooking", Integer.class);

    public final NumberPath<Integer> id = createNumber("id", Integer.class);

    public final com.mysema.query.sql.PrimaryKey<HotelTipoHabOverbooking> id = createPrimaryKey(id);

    public final com.mysema.query.sql.ForeignKey<HotelMoverbooking> hotelTipoHabOverbookingHotelMoverbookingFk = createForeignKey(fkIdHotelMoverbooking, "ID");

    public final com.mysema.query.sql.ForeignKey<HotelTipoHab> hotelTipoHabOverbookingHotelTipoHabFk = createForeignKey(Arrays.asList(fkCodtipohab, fkCodigohot), Arrays.asList("CODTIPOHAB", "FK_CODIGOHOT"));

    public QHotelTipoHabOverbooking(String variable) {
        super(HotelTipoHabOverbooking.class, forVariable(variable), "dbo", "HOTEL_TIPO_HAB_OVERBOOKING");
        addMetadata();
    }

    public QHotelTipoHabOverbooking(String variable, String schema, String table) {
        super(HotelTipoHabOverbooking.class, forVariable(variable), schema, table);
        addMetadata();
    }

    public QHotelTipoHabOverbooking(Path<? extends HotelTipoHabOverbooking> path) {
        super(path.getType(), path.getMetadata(), "dbo", "HOTEL_TIPO_HAB_OVERBOOKING");
        addMetadata();
    }

    public QHotelTipoHabOverbooking(PathMetadata<?> metadata) {
        super(HotelTipoHabOverbooking.class, metadata, "dbo", "HOTEL_TIPO_HAB_OVERBOOKING");
        addMetadata();
    }

    public void addMetadata() {
        addMetadata(cantidad, ColumnMetadata.named("CANTIDAD").withIndex(5).ofType(4).withSize(10).notNull());
        addMetadata(fecha, ColumnMetadata.named("FECHA").withIndex(6).ofType(93).withSize(23).withDigits(3).notNull());
        addMetadata(fkCodigohot, ColumnMetadata.named("FK_CODIGOHOT").withIndex(3).ofType(12).withSize(3).notNull());
        addMetadata(fkCodtipohab, ColumnMetadata.named("FK_CODTIPOHAB").withIndex(4).ofType(12).withSize(6).notNull());
        addMetadata(fkIdHotelMoverbooking, ColumnMetadata.named("FK_ID_HOTEL_MOVERBOOKING").withIndex(2).ofType(4).withSize(10).notNull());
        addMetadata(id, ColumnMetadata.named("ID").withIndex(1).ofType(4).withSize(10).notNull());
    }

}
```

The problem in that code:

``` java
public final NumberPath<Integer> id = createNumber("id", Integer.class);
public final com.mysema.query.sql.PrimaryKey<HotelTipoHabOverbooking> id = createPrimaryKey(id);
```

I´m using the maven plugin for the  code generation, shows something like:

``` shell
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project . . .
[ERROR] . . . /QHotelTipoHabOverbooking.java:[40,75] variable id is already defined in class QHotelTipoHabOverbooking
```

I know this can be easily fixed just using a proper name for that constraint, but changing the database is not and option, is a production version :S

--- a/querydsl-sql-codegen/src/main/java/com/mysema/query/sql/codegen/DefaultNamingStrategy.java
+++ b/querydsl-sql-codegen/src/main/java/com/mysema/query/sql/codegen/DefaultNamingStrategy.java
@@ -15,6 +15,7 @@ package com.mysema.query.sql.codegen;
 
 import java.util.Locale;
 
+import com.google.common.collect.ImmutableList;
 import com.mysema.query.codegen.EntityType;
 
 /**
@@ -76,7 +77,7 @@ public class DefaultNamingStrategy extends AbstractNamingStrategy {
         if (fkName.toLowerCase().startsWith("fk_")) {
             fkName = fkName.substring(3) + "_" + fkName.substring(0,2);
         }
-        return getPropertyName(fkName, entityType);
+        return escape(entityType, getPropertyName(fkName, entityType));
     }
 
     @Override
@@ -90,7 +91,13 @@ public class DefaultNamingStrategy extends AbstractNamingStrategy {
         if (pkName.toLowerCase().startsWith("pk_")) {
             pkName = pkName.substring(3) + "_" + pkName.substring(0,2);
         }
-        return getPropertyName(pkName, entityType);
+        String propertyName = getPropertyName(pkName, entityType);
+        for (String candidate : ImmutableList.of(propertyName, propertyName + "Pk")) {
+            if (!entityType.getEscapedPropertyNames().contains(candidate)) {
+                return candidate;
+            }
+        }
+        return escape(entityType, propertyName);
     }
 
     protected String normalizePropertyName(String name) {

--- a/querydsl-sql-codegen/src/test/java/com/mysema/query/sql/codegen/DefaultNamingStrategyTest.java
+++ b/querydsl-sql-codegen/src/test/java/com/mysema/query/sql/codegen/DefaultNamingStrategyTest.java
@@ -13,13 +13,12 @@
  */
 package com.mysema.query.sql.codegen;
 
-import static org.junit.Assert.assertEquals;
-
-import org.junit.Before;
-import org.junit.Test;
-
 import com.mysema.codegen.model.Types;
 import com.mysema.query.codegen.EntityType;
+import com.mysema.query.codegen.Property;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
 
 public class DefaultNamingStrategyTest {
 
@@ -86,11 +85,16 @@ public class DefaultNamingStrategyTest {
         assertEquals("refFooBar_", namingStrategy.getPropertyNameForForeignKey("REF_FOO_BAR_", entityModel));
     }
 
-
     @Test
     public void GetPropertyNameForPrimaryKey() {
         assertEquals("superiorPk", namingStrategy.getPropertyNameForPrimaryKey("pk_superior", entityModel));
         assertEquals("superiorPk", namingStrategy.getPropertyNameForPrimaryKey("PK_SUPERIOR", entityModel));
+    }
+
+    @Test
+    public void GetPropertyNameForPrimaryKey_Clash() {
+        entityModel.addProperty(new Property(entityModel, "id", Types.STRING));
+        assertEquals("idPk", namingStrategy.getPropertyNameForPrimaryKey("id", entityModel));
     }
 
     @Test

